spinn_utilities/make_tools/file_convertor.py

```diff
@@ -0,0 +1,453 @@
+import os
+import re
+import sys
+
+TOKEN = chr(30)  # Record Sperator
+
+STRING_REGEXP = re.compile('"([^"]|\\"|(""))*"')
+FORMATEXP = re.compile("%\d*(?:\.\d+)?[cdiksuxR]")
+LOG_END_REGEX = re.compile('\)(\s)*;')
+END_COMMENT_REGEX = re.compile("/*/")
+LOG_START_REGEX = re.compile(
+    "log_((info)|(error)|(debug)|(warning))(\s)*\(")
+
+# Status values
+NORMAL_CODE = 0
+COMMENT = NORMAL_CODE + 1
+IN_LOG = COMMENT + 1
+IN_LOG_CLOSE_BRACKET = IN_LOG + 1
+
+MINIS = {"log_info(": "log_mini_info(",
+         "log_error(": "log_mini_error(",
+         "log_debug(": "log_mini_debug(",
+         "log_warning(": "log_mini_warning("}
+
+LEVELS = {"log_info(": "[INFO]",
+          "log_error(": "[ERROR]",
+          "log_debug(": "[DEBUG]",
+          "log_warning(": "[WARNING]"}
+
+MAX_LOG_PER_FILE = 100
+
+
+class FileConvertor(object):
+    # __slots__ = [
+    #    "_dest", "_dest_basename", "_src", "_src_basename"]
+
+    # REGEXP = re.compile("%(?:\d+\$)?[dfsu]")
+
+    def __init__(self, src, dest, dict, range_start):
+        self._src = os.path.abspath(src)
+        self._dest = os.path.abspath(dest)
+        self._message_id = range_start
+        self._dict = dict
+
+    def run(self):
+        if not os.path.exists(self._src):
+            raise Exception("Unable to locate source {}".format(src))
+        dest_dir = os.path.dirname(os.path.realpath(self._dest))
+        if not os.path.exists(dest_dir):
+            os.makedirs(dest_dir)
+        self.convert_c()
```

Comment:
private method?

Reply:
YES! Especially as it goes POP if run a second time before first finishes.